src/container/srv_target.c

```diff
@@ -483,10 +485,22 @@ cont_aggregate_interval(struct ds_cont_child *cont, cont_aggregate_cb_t cb,
 			/* Don't sleep too long when there is space pressure */
 			msecs = 2ULL * 100;
 		}
+
+		if (param->ap_vos_agg)
+			cont->sc_vos_agg_active = 0;
+		else
+			cont->sc_ec_agg_active = 0;
+
 next:
 		if (dss_ult_exiting(req))
 			break;
 
+		/* sleep 18 seconds for EC aggregation ULT if the pool is in rebuilding,
+		 * if no space pressure.
+		 */
+		if (cont->sc_pool->spc_pool->sp_rebuilding && !param->ap_vos_agg && msecs != 200)
```

Comment:
is the msecs != 200 part of this conditional meant to refer to the branch above on line 484 that sets msecs = 2ULL * 100 when (sched_req_space_check(req) != SCHED_SPACE_PRESS_NONE) ?  Would it be clearer (and possibly future-proof) to reuse the latter in this check (or maybe a new boolean variable based on it), instead of the specific value of the msecs assignment on line 486?

Reply:
sched_req_space_check() with a few overhead. I considered to use another boolean for it to make it clearer, but as the func here is quite short and not difficult to understand, so I did not change as that.

Reply:
Do we still need to suspend EC aggregation when rebuilding in 2.8?